pkg/operator/controllers/autosizednodes/autosizednodes_controller.go

```diff
@@ -0,0 +1,123 @@
+package autosizednodes
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/coreos/ignition/v2/config/util"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+)
+
+type Reconciler struct {
+	client.Client
+	Scheme *kruntime.Scheme
+
+	log *logrus.Entry
+}
+
+const (
+	configName = "dynamic-node"
+)
+
+func NewReconciler(log *logrus.Entry, mgr ctrl.Manager) *Reconciler {
+	return &Reconciler{
+		Client: mgr.GetClient(),
+
+		log: log,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	var aro arov1alpha1.Cluster
+	var err error
+
+	err = r.Get(ctx, request.NamespacedName, &aro)
+	if err != nil {
+		r.log.Error("unable to fetch aro cluster")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	r.log.Infof("Config changed, autoSize: %t\n", aro.Spec.Features.ReconcileAutoSizedNodes)
+
+	// key is used to locate the object in the etcd
+	key := types.NamespacedName{
+		Name: configName,
+	}
+
+	if aro.Spec.Features.ReconcileAutoSizedNodes {
+		// create when needed
+		var config mcv1.KubeletConfig
+		err = r.Get(ctx, key, &config)
+		if err == nil {
+			// config exists
+			return ctrl.Result{}, nil
+		}
+		if client.IgnoreNotFound(err) != nil {
+			err = fmt.Errorf("could not fetch KubeletConfig: %v", err)
+			return ctrl.Result{}, err
+		}
+
+		config = makeConfig()
+
+		err = r.Create(ctx, &config, &client.CreateOptions{})
+		if err != nil {
+			err = fmt.Errorf("could not create KubeletConfig: %v", err)
+		}
+		return ctrl.Result{}, err
+	}
+
+	// defaults to deleting the config
+	config := makeConfig()
```

Comment:
The name and namespace is still required for deletion. I made this choice because it was already there. I updated the code to have only the required fields.